src/types.rs

```diff
@@ -211,6 +211,7 @@ pub struct State {
     pub hide_virtual_texts_on_insert: bool,
     pub echo_project_root: bool,
     pub enable_extensions: Option<HashMap<String, bool>>,
+    pub diagnostics_display_funcref: Option<String>,
```

Comment:
Maybe we should reuse the `diagnostics_list` config instead? And allow the values to be either `Quickfix`, `Location`, `Disabled` or a `Funcref` ? I think that makes more sense.

Reply:
Pushed another commit to remove that config and replace it for support for `funcref` on `LanguageClient_diagnosticsList`. Made some changes to how selectionUI works in those terms as well, for the sake of uniformity.